src/mito2/src/region/opener.rs

```diff
@@ -144,7 +150,7 @@ impl RegionOpener {
                 );
             }
         }
-        let options = RegionOptions::try_from(&self.options)?;
+        let options = self.options.take().unwrap();
```

Comment:
Is it safe to unwrap here?

Reply:
Yes, in this file all `unwrap` operations lack a safety comment. cc @evenyag

Reply:
No. The opener itself has a `Panics` note: It panics if the metadata is unset.

What's more, it isn't safe to unwrap here. Callers might forget to set the options. So we should not add the safety note here but document it in the `Panics` note.

Reply:
Updated the panics note